### PR TITLE
Create Demo Server Target

### PR DESCRIPTION
**Pull request by daming-lu**
Note the demo server target shares a large amount of common code with dev server in these files:

webpack.demo.config.js
demo-server.js
start_demo_server.sh

After I gain more experiences in webpack and npm, I will try to merge the common part.

Tracking it [here](https://github.com/PaddlePaddle/VisualDL/issues/432)

### CODE REVIEW

File: frontend/tool/webpack.demo.config.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+const webpack = require('webpack');
+const FriendlyErrorsPlugin = require('friendly-errors-webpack-plugin');
+let merge = require('webpack-merge');
+let baseWebpackConfig = require('./webpack.config');
+
+// add hot-reload related code to entry chunks
+Object.keys(baseWebpackConfig.entry).forEach(function (name) {
+    baseWebpackConfig.entry[name] = ['./tool/dev-client'].concat(baseWebpackConfig.entry[name]);
```
Comment (jetfuel) [inline frontend/tool/webpack.demo.config.js:9]: Not sure if we need to update the name from `./tool/dev-client` to ./tool/demo-client

File: frontend/tool/webpack.demo.config.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+});
+
+/**
+ * dev config
```
Comment (jetfuel) [inline frontend/tool/webpack.demo.config.js]: demo config

File: frontend/tool/demo-server.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+app.use(hotMiddleware);
+
+let uri = 'http://localhost:' + port;
+
+let _resolve;
+let readyPromise = new Promise(resolve => {
+    _resolve = resolve;
+});
+
+console.log('> Starting dev server...');
```
Comment (jetfuel) [inline frontend/tool/demo-server.js]: starting demo server.

File: frontend/tool/demo-server.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+process.env.NODE_ENV = 'demo'; // this is the only diff from dev server
+let devPort = 8999;
+let opn = require('opn');
+let express = require('express');
+let webpack = require('webpack');
+let proxyMiddleware = require('http-proxy-middleware');
+let webpackConfig = require('./webpack.dev.config');
```
Comment (jetfuel) [inline frontend/tool/demo-server.js]: Here is loading the incorrect config. Should update `./webpack.dev.config` to `./webpack.demo.config`

File: frontend/tool/webpack.demo.config.js (hunk 1 above)
Comment (daming-lu) [reply frontend/tool/webpack.demo.config.js:9]: We only have [dev-client.js](https://github.com/PaddlePaddle/VisualDL/blob/develop/frontend/tool/dev-client.js) which just includes hot-reload code.

**Review by jetfuel — APPROVED** [review]

**Pull request merged**